internal/controller/operatorconfigmap_controller.go

```diff
@@ -1075,3 +1089,74 @@ func parseTolerations(csiPluginTolerations string) ([]corev1.Toleration, error)
 	}
 	return tolerations, nil
 }
+
+func (c *OperatorConfigMapReconciler) removeNoobaa() error {
+	noobaa := &nbv1.NooBaa{}
+	noobaa.Name = "remote-noobaa"
+	noobaa.Namespace = c.OperatorNamespace
+
+	if err := c.get(noobaa); client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to get remote noobaa: %v", err)
+	} else if noobaa.UID != "" && noobaa.GetDeletionTimestamp().IsZero() {
+		index := slices.IndexFunc(
+			noobaa.GetOwnerReferences(),
+			func(ref metav1.OwnerReference) bool {
+				return ref.Kind == "StorageClient"
+			},
+		)
+		if index == -1 {
+			if err := c.delete(noobaa); err != nil {
+				return fmt.Errorf("failed to delete remote noobaa: %v", err)
+			}
+		}
+	}
+	return nil
+}
+
+func (c *OperatorConfigMapReconciler) removeNoobaaOperator() error {
+
+	nb := &metav1.PartialObjectMetadataList{}
+	nb.SetGroupVersionKind(nbv1.SchemeGroupVersion.WithKind("NooBaa"))
+	if err := c.list(nb); err != nil {
+		return fmt.Errorf("failed to list noobaa: %v", err)
+	}
+	if len(nb.Items) != 0 {
+		return nil
+	}
+
+	noobaaSubscription, err := c.getSubscriptionByPackageName("mcg-operator")
+	if client.IgnoreNotFound(err) != nil {
+		return err
+	}
+	if noobaaSubscription.UID != "" {
+		index := slices.IndexFunc(
+			noobaaSubscription.OwnerReferences,
+			func(ref metav1.OwnerReference) bool {
+				return ref.Kind == "Subscription" && ref.Name == "odf-operator"
+			},
+		)
+		if index != -1 {
+			return nil
+		}
+	}
```

Comment:
```suggestion
	if client.IgnoreNotFound(err) != nil {
		return err
	} if noobaaSubscription.UID == "" {
		return nil 
	} else {
		index := slices.IndexFunc(
			noobaaSubscription.OwnerReferences,
			func(ref metav1.OwnerReference) bool {
				return ref.Kind == "Subscription" && ref.Name == "odf-operator"
			},
		)
		if index != -1 {
			return nil
		}
	}
```

Reply:
dropped the else because of golang-ci lint
```Error: internal/controller/operatorconfigmap_controller.go:1133:9: indent-error-flow: if block ends with a return statement, so drop this else and outdent its block (revive)
  	} else {
  		index := slices.IndexFunc(
  			noobaaSubscription.OwnerReferences,
  			func(ref metav1.OwnerReference) bool {
  				return ref.Kind == "Subscription" && ref.Name == "odf-operator"
  			},
  		)
  		if index != -1 {
  			return nil
  		}
  	}```